bindgen-tests/tests/headers/function_macros.h

```diff
@@ -0,0 +1,15 @@
+// bindgen-flags: --macro-function "SIMPLE" --macro-function "INDIRECT_SIMPLE" --macro-function "f32 COMPLEX(u32)" --macro-function "f32 INDIRECT_COMPLEX(u32)" --macro-function "f32 CONDITIONAL_COMPLEX(bool, u32)"
```

Comment:
I have mixed feelings about the syntax for the `--macro-function` argument. Usually when we have these options where you need to match an item by name and need extra info for it we use the `--option=<ITEM_NAME>=<EXTRA_STUFF>` syntax like in the `--override-abi` case.

Maybe having `--macro-function=<MACRO_NAME>=<WRAPPER_TYPE>` is a bit more in style with the rest of bindgen.

This also opens the gates to `<MACRO_NAME>` being a regex in the future.

Reply:
Hm, the benefit of the the current representation is that is matches the macro definition with the rust types for argument names. For me that looks very readable, but we can try to match the existing command line style.

Did you have an idea for the `<EXTRA_STUFF>` syntax? Should we try to match the syntax used with the builder pattern like this?
`--macro-function=SIMPLE --macro-function=COMPLEX="f32 (u32)" --macro-function=INDIRECT_COMPLEX="f32 (u32)" --macro-function=CONDITIONAL_COMPLEX="f32 (bool, u32)"`